packages/babel-helpers/src/helpers.js

```diff
@@ -860,10 +860,27 @@ helpers.arrayWithHoles = helper("7.0.0-beta.0")`
 
 helpers.iterableToArray = helper("7.0.0-beta.0")`
   export default function _iterableToArray(iter) {
-    if (
-      Symbol.iterator in Object(iter) ||
-      Object.prototype.toString.call(iter) === "[object Arguments]"
-    ) return Array.from(iter);
+    // ES6 prototypes that use @@iterator
+    const prototypeWhiteList = [
+      String.prototype,
+      Array.prototype,
```

Comment:
`Array.isArray`

Reply:
`Array.isArray` isn't the same a prototype check. For example, `Array.isArray` would return `true` when checked against `Array.prototype` where this wouldn't. Also, `Array.isArray` is already checked prior to this function.